### DISCUSSION TITLE
How to get records that have been updated in `do_orm_execute` event?

### DISCUSSION THREAD

**Discussion by uralbash** [sqlalchemy/sqlalchemy#11939] [Usage Questions]
Is it possible to track which records were updated during a mass update of records?
```
def do_orm_execute(orm_execute_state: ORMExecuteState):
    if orm_execute_state.is_update:
        orm_execute_state.invoke_statement()
        updated_items = ???
```

**Comment by zzzeek** [comment]
the docs are not too clear about this at the moment, but if you run `session.execute(update(...))` and the update does not use ``returning()``, you will get [CursorResult](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.CursorResult) back which has rowcount:

```py
def do_orm_execute(orm_execute_state: ORMExecuteState):
    if orm_execute_state.is_update:
        result = orm_execute_state.invoke_statement()
        updated_items = result.rowcount
```
